### DISCUSSION TITLE
Best method for palette swapping of animated sprites?

### DISCUSSION THREAD

**Discussion by Lbones-ai** [HaxeFlixel/flixel#2353] [Q&A] [answered]
I was looking for palette swapping methods and I stumbled upon this:
`replaceColor(Color:FlxColor, NewColor:FlxColor, FetchPositions:Bool = false):Array<FlxPoint>`
> Replaces all pixels with specified Color with NewColor pixels. WARNING: very expensive (especially on big graphics) as it iterates over every single pixel.

I also found out about palette swapping via openFL shaders but I'm guessing is the same thing performance wise because in the API it says:
> Avoid changing it frequently as this is a costly operation.

There are better ways to do this?

**Comment by Gama11** [comment, marked as answer]
Shaders should perform fine. The comment refers to changing the shader _instance_ that's assigned to the sprite, which you shouldn't need to do.

**Reply by Lbones-ai** [reply]
Thanks, I should start get cracking with shaders then
